### PR TITLE
docs: Mention that a kv-store is optional with k8s.

### PR DESCRIPTION
**Pull request by jrajahalme**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10321)

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-docs-please

File: Documentation/install/system_requirements.rst [hunk 1]
```diff
@@ -109,13 +113,19 @@ linked, either choice is valid.
...
+nodes. The following Key-Value stores are currently supported:
 
 - etcd >= 3.1.0
 - consul >= 0.6.4
 
+Cilium can be used without a Key-Value store when CRD-based identity
+management is used with Kubernetes. This is the default for new Cilium
+installations. Larger clusters will perform better with a Key-Value
+store backed identity management instead, see :ref:`k8s_quick_install`
+for more.
```
Comment (joestringer) [inline Documentation/install/system_requirements.rst]:
```suggestion
for more details.
```

File: Documentation/install/system_requirements.rst [hunk 2]
```diff
@@ -109,13 +113,19 @@ linked, either choice is valid.
...
-security identities across all cluster nodes. The following Key-Value stores
-are currently supported:
+Cilium optionally uses a distributed Key-Value store to manage,
+synchronize and distribute security identities across all cluster
+nodes. The following Key-Value stores are currently supported:
 
 - etcd >= 3.1.0
 - consul >= 0.6.4
 
+Cilium can be used without a Key-Value store when CRD-based identity
```
Comment (joestringer) [inline Documentation/install/system_requirements.rst]:
```suggestion
Cilium can be used without a Key-Value store when CRD-based state
```

**Review by joestringer — CHANGES_REQUESTED** [review]
Minor nits, it's not just identity management that is done via CRD and the sentence wasn't complete.

File: Documentation/install/system_requirements.rst (hunk 1 above)
Comment (jrajahalme) [reply Documentation/install/system_requirements.rst]: Done

File: Documentation/install/system_requirements.rst (hunk 2 above)
Comment (jrajahalme) [reply Documentation/install/system_requirements.rst]: Done.

**Comment by jrajahalme** [comment]
test-docs-please

**Review by joestringer — APPROVED** [review]

**Pull request merged**